File: architect_zenghui/src/main/java/com/east/architect_zenghui/architect_21_designmode14/simple4/MainActivity.java
package com.east.architect_zenghui.architect_21_designmode14.simple4;

import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * Created by hcDarren on 2017/11/4.
 */

public class MainActivity extends AppCompatActivity {
    public TextView tv1;
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // ButterKnife.bind(this);
        MainActivity$$Binder binder = new MainActivity$$Binder();
        binder.bind(this);
    }
}

File: arouterpractice2/src/main/java/com/east/arouterpractice2/MainActivity.kt
package com.east.arouterpractice2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.alibaba.android.arouter.facade.annotation.Autowired
import com.alibaba.android.arouter.facade.annotation.Route
import com.alibaba.android.arouter.launcher.ARouter
import kotlinx.android.synthetic.main.arouter2_activity_main.*

@Route(path = "/kotlin/test")
class MainActivity : AppCompatActivity() {

    @Autowired
    @JvmField
    var name: String? = null
    @Autowired
    @JvmField
    var age: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        ARouter.getInstance().inject(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.arouter2_activity_main)

        tv_content.text = "name：$name-----age：$age"
//        tv_content.text = "name："
    }
}
